mp/src/game/client/momentum/ui/SettingsPanel/GamemodeSettingsPage.cpp

```diff
@@ -0,0 +1,99 @@
+#include "cbase.h"
+
+#include "GamemodeSettingsPage.h"
+
+#include "ienginevgui.h"
+#include "mom_shareddefs.h"
+#include "util/mom_util.h"
+
+#include <vgui_controls/ComboBox.h>
+#include <vgui_controls/CvarToggleCheckButton.h>
+#include <vgui_controls/Frame.h>
+#include <vgui_controls/TextEntry.h>
+
+#include "tier0/memdbgon.h"
+
+using namespace vgui;
+
+GamemodeSettingsPage::GamemodeSettingsPage(Panel *pParent) : BaseClass(pParent, "GamemodeSettings")
+{
+    // Rocket Jump controls
+    m_pRJParticlesBox = new ComboBox(this, "ParticlesBox", 3, false);
+    m_pRJParticlesBox->AddItem("#MOM_Settings_RJ_None", nullptr);
+    m_pRJParticlesBox->AddItem("#MOM_Momentum", nullptr);
+    m_pRJParticlesBox->AddItem("#MOM_Settings_RJ_TF2", nullptr);
+    m_pRJParticlesBox->AddActionSignalTarget(this);
+
+    m_pRJSoundsBox = new ComboBox(this, "SoundsBox", 3, false);
+    m_pRJSoundsBox->AddItem("#MOM_Settings_RJ_None", nullptr);
+    m_pRJSoundsBox->AddItem("#MOM_Momentum", nullptr);
+    m_pRJSoundsBox->AddItem("#MOM_Settings_RJ_TF2", nullptr);
+    m_pRJSoundsBox->AddActionSignalTarget(this);
+
+    m_pRJTrailBox = new ComboBox(this, "TrailBox", 3, false);
+    m_pRJTrailBox->AddItem("#MOM_Settings_RJ_None", nullptr);
+    m_pRJTrailBox->AddItem("#MOM_Momentum", nullptr);
+    m_pRJTrailBox->AddItem("#MOM_Settings_RJ_TF2", nullptr);
+    m_pRJTrailBox->AddActionSignalTarget(this);
+
+    m_pRJEnableTFRocketModel = new CvarToggleCheckButton(this, "EnableTFRocketModel", "#MOM_Settings_RJ_Enable_TF_RocketModel", "mom_rj_use_tf_rocketmodel");
+    m_pRJEnableTFViewModel = new CvarToggleCheckButton(this, "EnableTFViewModel", "#MOM_Settings_RJ_Enable_TF_ViewModel", "mom_rj_use_tf_viewmodel");
+    m_pRJEnableCenterFire = new CvarToggleCheckButton(this, "EnableCenterFire", "#MOM_Settings_RJ_Enable_Center_Fire", "mom_rj_center_fire");
+    m_pRJToggleRocketTrailSound = new CvarToggleCheckButton(this, "ToggleTrailSound", "#MOM_Settings_RJ_Trail_Sound_Enable", "mom_rj_trail_sound_enable");
+    m_pRJToggleRocketDecals = new CvarToggleCheckButton(this, "ToggleRocketDecals", "#MOM_Settings_RJ_Rocket_Decals_Enable", "mom_rj_decals_enable");
+
+    // Sticky Jump controls
+    m_pSJStickyDrawDelayEntry = new TextEntry(this, "DrawDelayEntry");
```

Comment:
Why not use CvarTextEntry?

Reply:
I frankly didn't know that was a thing, I'll use that instead.